### ISSUE TITLE
Wavebox becamos really slow after time. Some times crashes.

### ISSUE THREAD

**Issue by sfrique** [wavebox/waveboxapp#902]
* Wavebox Version: 4.5.10
* Operating System & Version: Archlinux
* (Linux only) Install method: aur
* Account type (if applicable): entire program

**Expected & actual behavior**
Program keep working normally.

**Steps to reproduce**
Let program open for a long period

**Is the bug persistent or intermittent?**
persistent

**Additional information**
On task monitor i didn't see any high cpu or memory, but program was really slow and most of times just is unusable.

**Comment by Thomas101** [comment]
@sfrique roughly what kind of time period are we talking about here? A day, week, longer etc... Which accounts do you have added into Wavebox?

**Comment by sfrique** [comment]
A day or a few days.

3 gmails
1 telegram
4 slacks
whatsapp
skype
hangouts chat
discord
messenger (facebook)

**Comment by sfrique** [comment]
> Thanks, there's nothing out of the ordinary there. We keep it running on a few machines here, and it doesn't seem to slow down. There's probably something that we've got configured differently, it would be good to know...
> 
> * Is it just Wavebox that slows up, or the rest of the system?

Just wavebox, it crashes sometimes.

> * What's the extent of the slowdown, for example...
>   * Slow to type 
>   * Slow to click on items and action them
>   * Slow to open Wavebox Settings

all the above, until it just not able to use it anymore and have to kill it.

> * If you go to Settings > Advanced, how do you have the following settings configured?
>   
>   * Hardware Acceleration

disabled

>   * Isolate Account Processes

disabled

>   * Sandboxing

enabled

>   * In process spellchecking

enable

> * Would you also be able to send a screenshot of Settings > Debug > Task Manager when...
>   
>   * You restart Wavebox

![restarted](https://user-images.githubusercontent.com/3711481/51247644-f4794600-1974-11e9-8636-cc3b43f92fbe.jpeg)


>   * Once it's slowed down

![hanged](https://user-images.githubusercontent.com/3711481/51247632-edeace80-1974-11e9-993d-d1b9de3a82f1.jpeg)

I think this pic when slowed down it was hanged and i could not used and had to kill it.

**Comment by Thomas101** [comment]
Thanks! The fact that none of that tabs are responding to the metrics service isn't good.

Do you remember why you disabled hardware acceleration? Just as a stab in the dark does re-enabling this make any difference?

**Comment by Thomas101** [comment]
Also, which Window Manager are you using?

**Comment by sfrique** [comment]
I disabled hardware acceleration because of lockscreen,  when back from lock screen it was hanged.

I am using i3 windows manager.

Will enabled to see if helps and there is no problem anymore.

I have to kill it many times a day know, it's  very annoying ='[

**Comment by sfrique** [comment]
Still getting very slow it's more frequent now.

Usually, it's not used for a while (30 min to 2 hours) and when I am back it's really slow to type, i just kill it and start over.

Sometime it even crashes.

Where can i find the logs so I let a tail open and see if anything weird appears.

**Comment by Thomas101** [comment]
Okay, I'll try running it on our i3 machine here to see if leaving to run results in a slowdown.

To grab some log info, there are a couple of things to try...

1. Run the app from the terminal, `/opt/wavebox/Wavebox`, and see if anything comes out from there. (If anything is going to come out, I would suspect it would be from here)
2. Run the app, then press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>J</kbd> to bring up the developer tools and click the console tab
3. You can also try logging stats to disk every 30 minutes to see if there's a runaway memory leak or similar. Do this from Settings > Debug, and Enable `Write app metrics`. This might give some more clues, as the task Manager was unable to query the processes for metrics in your screenshot above

Just some thoughts amoungst the team here, of things to try...

* If you try disabling Grammarly does that make any difference? I doubt it will, we're all using Grammarly here, but worth a try.
* Also with Grammarly disabled, does going to Settings > Advanced and disabling `In process spell-checking` change the behaviour? You might notice spellchecking is slightly janky or slower when using this setting.
* Which spellchecking language are you using at the moment? You can find this out under Settings > Language
* How much memory does your machine have and how much swap do you have configured?

Thanks!

**Comment by sfrique** [comment]
> Okay, I'll try running it on our i3 machine here to see if leaving to run results in a slowdown.
> 
> To grab some log info, there are a couple of things to try...
> 
> 1. Run the app from the terminal, `/opt/wavebox/Wavebox`, and see if anything comes out from there. (If anything is going to come out, I would suspect it would be from here)

There is some output a lot of those:
```

> 2. Run the app, then press Ctrl+Shift+J to bring up the developer tools and click the console tab
> 3. You can also try logging stats to disk every 30 minutes to see if there's a runaway memory leak or similar. Do this from Settings > Debug, and Enable `Write app metrics`. This might give some more clues, as the task Manager was unable to query the processes for metrics in your screenshot above
> 
> Just some thoughts amoungst the team here, of things to try...
> 
> * If you try disabling Grammarly does that make any difference? I doubt it will, we're all using Grammarly here, but worth a try.
> * Also with Grammarly disabled, does going to Settings > Advanced and disabling `In process spell-checking` change the behaviour? You might notice spellchecking is slightly janky or slower when using this setting.
> * Which spellchecking language are you using at the moment? You can find this out under Settings > Language
> * How much memory does your machine have and how much swap do you have configured?
> * If you launch the app, don't use it all but just leave it running does it still hang up within the 30-120 minutes?
> 
> Thanks!

**Comment by Thomas101** [comment]
Thanks @sfrique. Talking with the team here, we're starting to think it could be something to do with the file descriptor limit. Can you check the following...
* What file descriptor limit do you have configured? `ulimit -n`
* After running for a while, can you check how many file descriptors Wavebox has open. You should be able to do this with...
```bash
ls /proc/`ps -eF | grep "[/]opt/wavebox/Wavebox --mailto" | awk '{ print $2 }'`/fd | wc -l
```

Also, after a crash can you take a look to see if anything came out in syslog via
```bash
sudo grep -i wavebox /var/log/syslog
```

Thanks again

**Comment by sfrique** [comment]
Will do you command later, first this happened when running from terminal (it crashed over night)
```
....
(electron) 'memory' is deprecated and will be removed.
(electron) 'memory' is deprecated and will be removed.
(electron) 'memory' is deprecated and will be removed.
(electron) 'memory' is deprecated and will be removed.
(electron) 'memory' is deprecated and will be removed.
[METRICS][Thu Jan 24 2019 02:00:17 GMT-0200 (-02)] Log updated
Failed to get crash dump id.
Report Id: 
[1]    8614 segmentation fault (core dumped)  wavebox
```

**Comment by sfrique** [comment]
ulimit is 1024

as for your ps, my wavebox seems a litle different, but the main process has only 426 descriptors 
(This is from a clean start I will try to run after a while)
```
$ ls /proc/7948/fd/ | wc -l
426
$ ps -eF | grep 7948      
sfrique   7948  8016 15 481956 327300 3 11:05 pts/1    00:00:55 /usr/share/wavebox/Wavebox
sfrique   7950  7948  0 56285 41232   2 11:05 pts/1    00:00:00 /usr/share/wavebox/Wavebox --type=zygote --no-sandbox
sfrique   8021  7948  0 98272 103288  2 11:05 pts/1    00:00:02 /usr/share/wavebox/Wavebox --type=gpu-process --gpu-preferences=KAAAAAAAAACAAACAAQAAAAAAAAAAAGAAEAAAAAAAAAACAAAAAAAAAAgAAAAAAAAA --service-request-channel-token=2B41F2884AAA2DC906401FA211427683
sfrique  13977  8420  0  1578  2264   1 11:11 pts/9    00:00:00 grep --color 7948
```

There are logs crashes at journalctl
```
jan 24 02:16:48 sfrique-e7470 audit[8614]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=1 pid=8614 comm="Wavebox" exe="/usr/share/wavebox/Wavebox" sig=11 res=1
jan 24 02:16:48 sfrique-e7470 kernel: Wavebox[8614]: segfault at c4fbbb4e000 ip 00007efc6dd61ba5 sp 00007ffd6f3a3320 error 4 in libappindicator3.so.1.0.0[7efc6dd5e000+5000]
jan 24 02:16:48 sfrique-e7470 kernel: audit: type=1701 audit(1548303408.304:308): auid=1000 uid=1000 gid=1000 ses=1 pid=8614 comm="Wavebox" exe="/usr/share/wavebox/Wavebox" sig=11 res=1
jan 24 02:16:51 sfrique-e7470 systemd-coredump[6659]: Process 8614 (Wavebox) of user 1000 dumped core.
                                                      #6  0x000000000380f771 n/a (Wavebox)
                                                      #7  0x00000000038100da n/a (Wavebox)
                                                      #8  0x000000000246823b n/a (Wavebox)
                                                      #9  0x00000000037cabc7 n/a (Wavebox)
                                                      #10 0x0000000003778594 n/a (Wavebox)
                                                      #11 0x00000000037ea00e n/a (Wavebox)
                                                      #12 0x00000000037ea677 n/a (Wavebox)
                                                      #13 0x00000000037eaed9 n/a (Wavebox)
                                                      #14 0x00000000037db865 n/a (Wavebox)
                                                      #15 0x0000000003cf5001 n/a (Wavebox)
                                                      #16 0x0000000003cf4e03 n/a (Wavebox)
                                                      #17 0x0000000003d1fc92 n/a (Wavebox)
                                                      #18 0x0000000003cf0c20 n/a (Wavebox)
                                                      #19 0x0000000003dbe9c8 n/a (Wavebox)
                                                      #20 0x00000000036d3634 n/a (Wavebox)
                                                      #21 0x0000000003dbeb04 n/a (Wavebox)
                                                      #22 0x0000000004eae5a6 main (Wavebox)
                                                      #24 0x00000000011d002a _start (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037fb2fd n/a (Wavebox)
                                                      #2  0x00000000037fa55c n/a (Wavebox)
                                                      #3  0x00000000037eb580 n/a (Wavebox)
                                                      #4  0x00000000037db865 n/a (Wavebox)
                                                      #5  0x00000000037c6e24 n/a (Wavebox)
                                                      #6  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037fb2fd n/a (Wavebox)
                                                      #2  0x00000000037fa55c n/a (Wavebox)
                                                      #3  0x00000000037eb580 n/a (Wavebox)
                                                      #4  0x00000000037db865 n/a (Wavebox)
                                                      #5  0x00000000037c6e24 n/a (Wavebox)
                                                      #6  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x0000000004efabef n/a (Wavebox)
                                                      #2  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037fb2fd n/a (Wavebox)
                                                      #2  0x00000000037fa55c n/a (Wavebox)
                                                      #3  0x00000000037eb562 n/a (Wavebox)
                                                      #4  0x00000000037db865 n/a (Wavebox)
                                                      #5  0x00000000037c6e24 n/a (Wavebox)
                                                      #6  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x0000000003789246 n/a (Wavebox)
                                                      #2  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037657ff n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x00000000037eabd1 n/a (Wavebox)
                                                      #3  0x00000000037db865 n/a (Wavebox)
                                                      #4  0x00000000037c6e24 n/a (Wavebox)
                                                      #5  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037657ff n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x0000000003765055 n/a (Wavebox)
                                                      #2  0x000000000358b6e8 n/a (Wavebox)
                                                      #3  0x00000000037c6632 n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037fb2fd n/a (Wavebox)
                                                      #2  0x00000000037fa55c n/a (Wavebox)
                                                      #3  0x00000000037eb562 n/a (Wavebox)
                                                      #4  0x00000000037db865 n/a (Wavebox)
                                                      #5  0x00000000037c6e24 n/a (Wavebox)
                                                      #6  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037657ff n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d3cc n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x0000000003765055 n/a (Wavebox)
                                                      #2  0x0000000004411408 n/a (Wavebox)
                                                      #3  0x0000000004411309 n/a (Wavebox)
                                                      #4  0x0000000004413a50 n/a (Wavebox)
                                                      #5  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d3cc n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000039fa6b2 n/a (Wavebox)
                                                      #2  0x00000000037c6632 n/a (Wavebox)
                                                      #3  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037fb2fd n/a (Wavebox)
                                                      #2  0x00000000037fa55c n/a (Wavebox)
                                                      #3  0x00000000037eb562 n/a (Wavebox)
                                                      #4  0x00000000037db865 n/a (Wavebox)
                                                      #5  0x0000000003cfe034 n/a (Wavebox)
                                                      #6  0x0000000003cfe0af n/a (Wavebox)
                                                      #7  0x00000000037c6e24 n/a (Wavebox)
                                                      #8  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #1  0x00000000037658e4 n/a (Wavebox)
                                                      #2  0x000000000376a46b n/a (Wavebox)
                                                      #3  0x000000000376d62a n/a (Wavebox)
                                                      #4  0x00000000037cab21 n/a (Wavebox)
                                                      #3  0x0000000004f4b670 _ZN4atom12NodeBindings17EmbedThreadRunnerEPv (Wavebox)
```

**Comment by Thomas101** [comment]
That's really useful! That crash looked like it came out of libAppIndicator. If you go to Settings > Tray and disable `Show Tray`. Then restart the app, does it change the frequency of errors?

**Comment by Thomas101** [comment]
Thanks for that. There are a couple of users with crashing on Linux and I was starting to think that what you're seeing is related to that - but as we've been doing some debugging I'm starting to think less so.

Did disabling the tray make any difference?

**Comment by sfrique** [comment]
Lets wait today and see.
I had to kill it yesterday as well, but it seems it is taking longer to need to be killed.

**Comment by Thomas101** [comment]
@sfrique we've just released a new version (4.7.11) of Wavebox to [beta channel](https://wavebox.io/kb/beta-channel) with an updated version of Chromium. We're hoping this will fix the issue you've been seeing. It would be great if you could give this a try and see if it's fixed for you. Thanks :)

**Comment by sfrique** [comment]
@Thomas101 Thanks for the update I am now at 4.7.0 and it seems the problem went away.

I don't see more crashes nor white screens. (At least not that i can remember)

Thanks, I am closing the issue!
